### DISCUSSION TITLE
[C4GT][ULP] Consent template manager

### DISCUSSION THREAD

**Discussion by snehal0904** [Sunbird-RC/community#680] [Ideas & Enhancements]
### Project Details
The consent template manager will be an admin service, it will get used to create consent dynamic templates. ULP will use this service to get consent from the teacher/principal before granting the students' credentials, approving the student's claim, etc. 

### Features to be implemented

- [ ] Build APIs to create, update, delete and list templates
- [ ] Build APIs to validate and add tags in the template
- [ ] Build APIs to render template
- [ ] Build UI interface for the template manager

#### Complexity
Medium

### Skills Required
NodeJS Angular

### Name of Mentors:
Snehal Patil

### Project size
5 Weeks

### Product Set Up
The project has a docker file to setup will add a link shortly but this will be a separate service and needs to be done from scratch.

### Acceptance Criteria

- [ ]  e2e Test Caes
- [ ]  OpenAPI Spec/Postman Collection
- [ ] Dockerfile for this module

### Milestone
Every document type supported is a milestone.

C4GT
This issue is nominated for Code for GovTech (C4GT) 2023 edition.
C4GT is India's first annual coding program to create a community that can build and contribute to global [Digital Public Goods](https://www.codeforgovtech.in/digitalpublicgoods). If you want to use Open Source GovTech to create impact, then this is the opportunity for you! More about C4GT here: https://codeforgovtech.in/

**Comment by ak-ironix** [comment]
Hello Snehal, I am a 3rd year student at LNM Institute of Information Technology, I have worked on Angular and NodeJs having made a few CRUD application on this tech stack. I think I would be a great fit for this project and would like to contribute towards the same under C4GT.

**Comment by simran142002** [comment]
Hello @snehal0904 is there reference link for the repository?

**Comment by DishaJain17** [comment]
Hi  @snehal0904  ,My name is Disha Jain, and I'm excited to be a part of this project. I'm currently looking for open issues to contribute to, and I would greatly appreciate it if you could suggest some for me to work on.

I have a strong technical background, particularly in the MERN stack, Java, C++, Linux, Git, and I also have a solid understanding of basics of YAML, Docker, and Kubernetes. I'm eager to utilize my skills and knowledge to contribute meaningfully to this project.Also i have a knowledge of web scraping and payment integration .

Resume:https://drive.google.com/file/d/1-mPdea8ZHQv7B19y0v8CDHl3rEJsowie/view?usp=drivesdk

Intro:
https://drive.google.com/file/d/1-8wHZN-0Ew3aojcU3xJm9MNUoWWrAoAI/view?usp=drivesdk

Blogs:https://hashnode.com/@disha390

LinkedIn: https://www.linkedin.com/in/disha-jain-90a554229

Thank you in advance for your support and recommendations. I'm looking forward to hearing from you soon!

Best Regards
Disha Jain

**Comment by Kapil650** [comment]
@snehal0904 This project is interesting, and I am looking forward to contributing to this work in the C4GT 2023.
My Name is Kapil, MCA final year CRSU student with knowledge of Angular, React and Node as provided in curriculum of MCA and also done certifications to enhance my Knowlege about MERN Stack.
